pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -205,10 +205,17 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 	}
 
 	for _, pod := range dswp.podManager.GetPods() {
-		if dswp.podStateProvider.ShouldPodContainersBeTerminating(pod.UID) {
+		// Keep consistency of adding pod during reconstruction
+		// https://github.com/kubernetes/kubernetes/issues/113979
+		if dswp.hasAddedPods && dswp.podStateProvider.ShouldPodContainersBeTerminating(pod.UID) {
+			continue
+		}
+
+		if !dswp.hasAddedPods && dswp.podStateProvider.ShouldPodRuntimeBeRemoved(pod.UID) {
 			// Do not (re)add volumes for pods that can't also be starting containers
```

Comment:
This comment should be moved to the first condition.

For this condition, we can add a comment along the lines of "When kubelet restarts, we need to add pods to dsw if there's a possibility that the container may still be running."